### DISCUSSION TITLE
High GPU memory consumption at inference with VITS

### DISCUSSION THREAD

**Discussion by tekinek** [coqui-ai/TTS#862] [General Q&A]
Thanks for your great work!

I have been a fan of tacotron implementation in Mozilla TTS, especially DDC. It is so far the most stable framework I guess. But the 'need-for-speed' forces me to try other frameworks, and thanks to the coqui-tts; most of the fast frameworks are implemented.

I have a few problems to solve:
1) VITS gives me satisfactory result, but it consumes much more GPU memory (or RAM) than other frameworks. it is almost impossible to synthesis a long sentence (e.g. 300 chars) without an error of CUDA out of Memory. In case of Tac2+Melgan, the consumption never exceeds 3GB.  Is this a bug or expected behaviours? is there any solution except keeping the sentence short?

2) With the same audio settings, tacotron2-produced mels works well with my exisitng vocoder (MB-Melgan), but the one with the FastPitch and SpeedySpeech outputs are complete mass, worst than grifflim. Do FastPitch and SpeedySpeech need a universal vocoder? or something may wrong in my setting?

3) I say Tacotron is stable becuase it producess the speech with relativly consistant prosody and tempo regardless of the length of sentences. But, the FastPich and SpeedySpeech ( and FastSpeech2 in other repo) not only fails to synthesis short content like one or two words, the tempo of the speech changes obviously when the length of sentence changes. I know the most of the sentences are long (>100 chars) in my 25-hours dataset, but Is there any solusion for this?

Thanks again!

**Comment by erogol** [comment]
300 chars = 600 chars with VITS since it adds a blank character after every other character. So try splitting your text into sentences and pass the separately to the model

**Reply by tekinek** [reply]
Thanks! Splitting the text into sentences prevents OOM error. However, memory consumtion of VIST seems linearly increasing by the legnth of text. In my case, 150 chars (300 with blanks) takes 6GB GPU memory. Sorry, but is there any solution to make it more memory efficient? so that I can run multiple instances in a single GPU card. (3~4 Tac2+Melgan instances work fine in a GPU of 11GB memory without limitation on sentense length)

**Reply by erogol** [reply]
Currently, there is no way to get better than this.
